api/config/v1alpha1/shared_types.go

```diff
@@ -56,6 +58,11 @@ type KubernetesDeploymentSpec struct {
 	// +optional
 	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
 
+	// Image specifies the image used by the container.
```

Comment:
```suggestion
	// Image specifies the EnvoyProxy container image to be used, instead of the default image.
	// The default image details are mentioned in https://gateway.envoyproxy.io/latest/intro/compatibility.html
	// Note: the version of the image must match the version of the default Envoy image
```